src/vs/workbench/contrib/chat/browser/chatViewPane.ts

```diff
@@ -317,4 +351,56 @@ export class ChatViewPane extends ViewPane implements IViewWelcomeDelegate {
 			(this.viewState as Record<string, unknown>)[key] = value;
 		}
 	}
+
+	private updateViewTitle(model?: IChatModel): void {
+		model = model ?? this._widget.viewModel?.model;
+		const title = (model && model.hasCustomTitle) ? model.title : localize('chat', "Chat");
+
+		this.updateTitle(localize('chat', "Chat"));
+		if (model && model.hasCustomTitle) {
+			this.setSecondaryTitle(title);
+		} else if (model && model.getRequests().length > 0) {
+			this.setSecondaryTitle(localize('generatingTitle', "Generating title..."));
```

Comment:
I have some saved chats that didn't get a custom title generated. That can happen for different reasons, like the request might fail, or maybe I was just debugging, but then we end up showing "generating title" forever. In that case we still show the request text as the chat title, we should show that here too. This state could probably look nicer, maybe it can show the request text instead, or "new chat", what do you think?